### PR TITLE
Support tilde_path metadata

### PR DESCRIPTION
**Pull request by live1206**
[AB#268911](https://dev.azure.com/ceapex/Engineering/_workitems/edit/268911)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/6269)

### CODE REVIEW

File: src/docfx/build/link/LinkResolver.cs [hunk 1]
```diff
@@ -218,8 +221,13 @@ internal class LinkResolver
 
             if (relativePath.StartsWith("~/") || relativePath.StartsWith("~\\"))
             {
-                // Treat ~/ as path relative to docset
-                pathToDocset = new PathString(relativePath.Substring(2).TrimStart('/', '\\'));
+                var (_, metadata) = _metadataProvider.GetMetadata(referencingFile);
+                pathToDocset = new PathString(Path.GetRelativePath(
```
Comment (yufeih) [inline src/docfx/build/link/LinkResolver.cs:225]: Consider only do the `Path.Combine` here and move line 238 - line 242 to line 244

Comment (live1206) [reply src/docfx/build/link/LinkResolver.cs:225]: right

**Pull request merged**